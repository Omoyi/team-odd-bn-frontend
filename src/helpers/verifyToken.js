import jwt from 'jsonwebtoken';

const verifyToken = (token) => jwt.verify(token, process.env.JWT_KEY, (err, decoded) => {
  if (err) {
    return false;
  }
  return decoded;
});

export default verifyToken;
