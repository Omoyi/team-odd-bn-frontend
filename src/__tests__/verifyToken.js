import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import dotenv from 'dotenv';
import verifyToken from '../helpers/verifyToken';
import ProtectRoutes from '../Routers/protectRoutes';
import generateToken from '../helpers/generateToken';
import mockData from '../__mocks__/fileMock';

// mock browser route
import rrd from 'react-router-dom';

rrd.BrowserRouter = ({ children }) => <div>{ children }</div>;


dotenv.config();

const { jwtTokenMocks } = mockData;
const { JWT_KEY } = process.env;

describe('Verify user token', () => {
  let wrapper;
  it('Expect protect components to render ', () => {
    wrapper = shallow(<ProtectRoutes {...jwtTokenMocks} />);
    expect(wrapper.props().token).toBe('I am token');
    expect(wrapper.props().error).toEqual({ message: 'Invalid token' });
  });

  it('Expect token to pass', () => {
    const randomToken = generateToken({ name: 'John' }, JWT_KEY);
    const result = verifyToken(randomToken);
    expect(result.name).toEqual('John');
  });
});


describe('Expect to check all tokens', () => {
  const wrapper = render(
    <MemoryRouter initialEntries={['/dashboard']}>
      <ProtectRoutes {...jwtTokenMocks} />
    </MemoryRouter>,
  );

  it('Expect token to pass', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
