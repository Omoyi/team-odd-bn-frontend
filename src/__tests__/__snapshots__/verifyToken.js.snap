// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to check all tokens Expect token to pass 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
